packages/core/src/components/icon/icon.tsx

```diff
@@ -142,8 +145,16 @@ export class Icon extends AbstractPureComponent2<IconProps & Omit<React.HTMLAttr
                 className: classes,
                 title: htmlTitle,
             },
-            <svg fill={color} data-icon={icon} width={size} height={size} viewBox={viewBox}>
-                {title && <desc>{title}</desc>}
+            <svg
+                fill={color}
+                data-icon={icon}
+                width={size}
+                height={size}
+                viewBox={viewBox}
+                aria-labelledby={title ? titleId : undefined}
```

Comment:
why not just use [`aria-label`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label)? the `<title>` is not visible on the page anyway, so I'm not sure that `aria-labelledby` is the best approach here.